sqlmesh/core/reference.py

```diff
@@ -53,18 +53,66 @@ def name(self) -> str:
 
 class ReferenceGraph:
     def __init__(self, models: t.Iterable[Model]):
-        self._model_refs: t.DefaultDict[str, t.Dict[str, Reference]] = defaultdict(dict)
-        self._ref_models: t.DefaultDict[str, t.Dict[str, str]] = defaultdict(dict)
+        self._model_refs: t.Dict[str, t.Dict[str, Reference]] = {}
+        self._ref_models: t.Dict[str, t.Dict[str, str]] = {}
```

Comment:
What's the point of setting both a key and a value of the nested dict to the same value (model name)? Is this because of dict's ordering guarantees?

Reply:
In other words, shouldn't this be similar to `_dim_models`?

Reply:
hm yea looks like oversight